apps/src/templates/studioHomepages/ReCaptchaValidationDialog.jsx

```diff
@@ -0,0 +1,101 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import i18n from '@cdo/locale';
+import Button from '../Button';
+import BaseDialog from '../BaseDialog';
+import Spinner from '../../code-studio/pd/components/spinner';
+
+const styles = {
+  dialog: {
+    padding: '20px'
+  }
+};
+
+export default class ReCaptchaValidationDialog extends React.Component {
```

Comment:
How much work would it be to make this a generic captcha component that could be used in other scenarios?  I could see that being useful in the future.

Reply:
PR for generic captcha component [here](https://github.com/code-dot-org/code-dot-org/pull/38032).